### PR TITLE
ci pipeline tests for ortmodule

### PR DESCRIPTION
**Pull request by baijumeswani**
**Description**:

Add tests for the ortmodule-api for the orttraining-ortmodule ci pipeline.

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        assert parameter_value.device.type == 'cpu'
+
+    model = model.cuda()
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cuda'
+
+def test_model_cpu():
+    device = 'cuda'
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:172]: Ha!

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cpu'
+
+    model = model.to(device)
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cuda'
+
+def test_model_to_tensor():
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:203]: you can parameterize to test several valid combinations, such as cpu, cuda, cuda:0

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        assert parameter_value.device.type == 'cuda'
+
+    model = model.cpu()
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cpu'
+
+def test_model_to_device():
+    device = 'cuda'
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:188]: you can parameterize to test several valid combinations, such as cpu, cuda, cuda:0

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    model = ORTModule(model)
+    x = torch.randn(N, D_in, device=device)
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cpu'
+
+    model = model.to(x)
+    model(x)
+
+    for _, parameter_value in model.named_parameters():
+        assert parameter_value.device.type == 'cuda'
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:217]: dont forget the new line at the end :)

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]
I missed a test that does model.to('cuda') and then model.to('cpu') to force the re-export

The to() and cuda() and cpu() override always have a condition to check whether the device was changed between calls

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py (hunk 4 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:217]: I think there is a new line already. GitHub doesn't show it well. I think the absence of a new line is indicated by the red circle by GitHub.

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py (hunk 1 above)
Comment (baijumeswani) [reply orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:172]: The ```device``` here is just for using a string literal. Basically, the model starts off as a cuda one. And then I call ```cpu``` on it and check whether the named_parameters have been migrated to cpu. I can leave a comment for this.

**Comment by baijumeswani** [comment]
Added  test that does ```to(device)``` and then goes back to original device using ```to(original_device)```. Made sure to check ```_require_export``` when doing so.

**Review by thiagocrepaldi — APPROVED** [review]

**Comment by baijumeswani** [comment]
Thank you

**Pull request merged**